### ISSUE TITLE
Clustering : Markers are not updated

### ISSUE THREAD

**Issue by InsideApp-OBO** [googlemaps/android-maps-compose#269]
#### Environment details

Library version: Tested with Maps Compose 2.11.0

#### Steps to reproduce (List with the same reference)

  1. Display N markers on the map with **_Clustering_** composable.
  2. Change the content of the list with totally different markers.
  3. Markers are not updated due to

```
    LaunchedEffect(items) {
        clusterManager.clearItems()
        clusterManager.addItems(items)
    }
```

#### Steps to reproduce (With a new list)

  1. Display N markers on the map with **_Clustering_** composable.
  2. Change the content of the list with totally different markers.
  3. Make a copy of the list with **_toList()_** function
  4. Markers are updated only when the zoom level changes.

#### Code example

```
@OptIn(MapsComposeExperimentalApi::class)
@Composable
fun GoogleMapClustering() {

    val items = remember { mutableStateListOf<MyItem>() }

    val cameraPositionState = rememberCameraPositionState {
        position = CameraPosition.fromLatLngZoom(singapore, 8f)
    }

    LaunchedEffect(cameraPositionState) {
        snapshotFlow { cameraPositionState.isMoving }
            .collect { isMoving ->
                if (!isMoving) {
                    items.apply {
                        clear()
                        for (i in 1..10) {
                            val position = LatLng(
                                singapore.latitude + Random.nextFloat(),
                                singapore.longitude + Random.nextFloat(),
                            )
                            add(MyItem(position, "Marker $i", "Snippet"))
                        }
                    }
                }
            }
    }

    Box(modifier = Modifier.fillMaxSize()) {
        GoogleMap(
            modifier = Modifier.fillMaxSize(),
            cameraPositionState = cameraPositionState
        ) {
            Clustering(
                items = items//.toList() // Force new list to avoid Clustering LaunchedEffect(items)
            )
        }
        Text(text = "Items : ${items.map { it.itemPosition }}")
    }
}
```

How to dynamically update the markers when the map is moved ?
(Ex : API Rest call with ViewModel)

Thank you for all your work !

**Comment by wangela** [comment]
If you would like to upvote the priority of this issue, please comment below or react with :+1: so we can see what is popular when we triage.
--------

@InsideApp-OBO Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Check the issue tracker](https://developers.google.com/maps/support/#issue_tracker) - bugs and feature requests for Google Maps Platform APIs and SDKs
- [Open a support case](https://developers.google.com/maps/support#contact-maps-support) - Get 1:1 support in Cloud Console.
- [Discord](https://discord.gg/fHTDmZe9En) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag


_This is an automated message, feel free to ignore._

**Comment by DSteve595** [comment]
Confirmed issue, thanks for the report and detailed info!

**Comment by NielsLindberg** [comment]
Needed for any filtering functionality to work without having to zoom after applying filters. With the old MapEffect solution, I had to call .cluster() as well: 
 clusterManager.clearItems()
 clusterManager.addItems(page.places)
 clusterManager.cluster()

**Comment by DSteve595** [comment]
A fix is up for review in #279.
